Compilation fails on ArchLinux

<!--- 
Please try to only report a bug if it happens with the latest version
The latest version can be seen by checking the ChangeLog: https://owncloud.org/changelog/desktop/

For support try: https://central.owncloud.org/c/help/desktop-file-sync
--->


### Expected behaviour
Running `make` shouldn't display build errors

### Actual behaviour
Running `make` displays the following error:

    /tmp/client_theming/client/src/libsync/theme.cpp:33:39: fatal error: /tmp/client_theming/build-1/../nextcloudtheme/nextcloudtheme.h: No such file or directory
    #  include INCLUDE_FILE(THEME_INCLUDE) 
                                                                            ^

### Steps to reproduce
1. Install ArchLinux
2. Follow build directions, a verbose account of the steps I took is below.

#### Checkout the repo
```
git clone https://github.com/nextcloud/client_theming.git
Cloning into 'client_theming'...
remote: Counting objects: 623, done.
remote: Total 623 (delta 0), reused 0 (delta 0), pack-reused 623
Receiving objects: 100% (623/623), 1.24 MiB | 0 bytes/s, done.
Resolving deltas: 100% (186/186), done.
```

#### Update the submodules

```
[tmp]$ cd client_theming/

[client_theming]$ git submodule update --init --recursive
Submodule 'client' (https://github.com/owncloud/client.git) registered for path 'client'
Cloning into '/tmp/client_theming/client'...
Submodule path 'client': checked out 'eaeed08544d1d7f4031d28a8e1bd9dd5e08a60fd'
Submodule 'binary' (git://github.com/owncloud/owncloud-client-binary.git) registered for path 'client/binary'
Submodule 'doc/ocdoc' (https://github.com/owncloud/documentation) registered for path 'client/doc/ocdoc'
Submodule 'src/3rdparty/libcrashreporter-qt' (git://github.com/dschmidt/libcrashreporter-qt.git) registered for path 'client/src/3rdparty/libcrashreporter-qt'
Submodule 'src/3rdparty/qtkeychain' (https://github.com/frankosterfeld/qtkeychain.git) registered for path 'client/src/3rdparty/qtkeychain'
Submodule 'src/3rdparty/qtmacgoodies' (https://github.com/guruz/qtmacgoodies.git) registered for path 'client/src/3rdparty/qtmacgoodies'
Cloning into '/tmp/client_theming/client/binary'...
Cloning into '/tmp/client_theming/client/doc/ocdoc'...
Cloning into '/tmp/client_theming/client/src/3rdparty/libcrashreporter-qt'...
Cloning into '/tmp/client_theming/client/src/3rdparty/qtkeychain'...
Cloning into '/tmp/client_theming/client/src/3rdparty/qtmacgoodies'...
Submodule path 'client/binary': checked out '0d89ac776654472a19c099bdfe410d3a2d930497'
Submodule path 'client/doc/ocdoc': checked out '2fdd8b2833ea1db7da08a61afda665eb6ecaa017'
Submodule path 'client/src/3rdparty/libcrashreporter-qt': checked out 'e8fffe61e7c94ce88e59b80579754c4a46da65ea'
Submodule path 'client/src/3rdparty/qtkeychain': checked out '08be0593e100a041585bcc704a1bbc5a812a1682'
Submodule path 'client/src/3rdparty/qtmacgoodies': checked out '044580c32837edba05a055aabca27245939454eb'
```

#### Run cmake
```
[client_theming]$ mkdir build-linux
[client_theming]$ cd build-linux
[build-linux]$ cmake -D OEM_THEME_DIR=`pwd`/../nextcloudtheme ../client
-- The C compiler identification is GNU 6.2.1
-- The CXX compiler identification is GNU 6.2.1
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Build of crashreporter disabled.
-- GIT_SHA1 eaeed08544d1d7f4031d28a8e1bd9dd5e08a60fd
-- Found OpenSSL: /usr/lib64/libssl.so;/usr/lib64/libcrypto.so (found suitable version "1.0.2j", minimum required is "1.0.0") 
-- Could NOT find INOTIFY (missing:  INOTIFY_LIBRARY_DIR) 
-- Found Sphinx: /usr/bin/sphinx-build  
-- Could NOT find PdfLatex (missing:  PDFLATEX_EXECUTABLE) 
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.1") 
-- Checking for module 'sqlite3'
--   Found sqlite3, version 3.15.0
-- Found SQLite3: /usr/lib64/libsqlite3.so (Required is at least version "3.8.0") 
-- Found ZLIB: /usr/lib64/libz.so (found version "1.2.8") 
-- Performing Test WITH_FPIC
-- Performing Test WITH_FPIC - Success
-- Performing Test WITH_STACK_PROTECTOR
-- Performing Test WITH_STACK_PROTECTOR - Success
-- Looking for iconv.h
-- Looking for iconv.h - found
-- Looking for sys/iconv.h
-- Looking for sys/iconv.h - not found
-- Looking for iconv
-- Looking for iconv - found
-- Checking prototype iconv for HAVE_ICONV_CONST - False
-- Found Iconv: /usr/include  
-- Found SQLite3: /usr/lib64/libsqlite3.so;/usr/lib64/libsqlite3.so (Required is at least version "3.8.0") 
-- Looking for argp.h
-- Looking for argp.h - found
-- Looking for clock_gettime in rt
-- Looking for clock_gettime in rt - found
-- Looking for dlopen in dl
-- Looking for dlopen in dl - found
-- Looking for asprintf
-- Looking for asprintf - found
-- Looking for fnmatch
-- Looking for fnmatch - found
-- Looking for timegm
-- Looking for timegm - found
-- Looking for strerror_r
-- Looking for strerror_r - found
-- Looking for utimes
-- Looking for utimes - found
-- Looking for lstat
-- Looking for lstat - found
-- Found Qt5 core, checking for further dependencies...
-- Using Qt 5!
-- Found Qt5Keychain: /usr/lib64/libqt5keychain.so  
-- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY
-- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY - Success
-- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY
-- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY - Success
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR - Success
CMake Warning (dev) at /usr/share/ECM/modules/ECMFindModuleHelpers.cmake:133 (message):
  Your project should require at least CMake 2.8.12 to use FindKF5.cmake
Call Stack (most recent call first):
  /usr/share/ECM/find-modules/FindKF5.cmake:52 (ecm_find_package_version_check)
  shell_integration/CMakeLists.txt:11 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning at /usr/share/ECM/find-modules/FindKF5.cmake:74 (find_package):
  Could not find a package configuration file provided by "KF5KIO" (requested
  version 5.16) with any of the following names:

    KF5KIOConfig.cmake
    kf5kio-config.cmake

  Add the installation prefix of "KF5KIO" to CMAKE_PREFIX_PATH or set
  "KF5KIO_DIR" to a directory containing one of the above files.  If "KF5KIO"
  provides a separate development package or SDK, be sure it has been
  installed.
Call Stack (most recent call first):
  shell_integration/CMakeLists.txt:11 (find_package)


-- Could NOT find KF5KIO: found neither KF5KIOConfig.cmake nor kf5kio-config.cmake 
-- Could NOT find KF5 (missing:  KIO) (Required is at least version "5.16")
Dolphin plugin disabled: KDE Frameworks 5.16 not found
-- Found Doxygen: /usr/bin/doxygen (found version "1.8.12") 
-- Configuring done
-- Generating done
-- Build files have been written to: /tmp/client_theming/build-linux
```

Note: I didn't install KIO as it would bring in a ton of KDE libs that I don't need/want.

#### Run make
```
[build-linux]$ make
Scanning dependencies of target cstdlib
[  0%] Building C object csync/src/std/CMakeFiles/cstdlib.dir/c_alloc.c.o
[  0%] Building C object csync/src/std/CMakeFiles/cstdlib.dir/c_path.c.o
[  1%] Building C object csync/src/std/CMakeFiles/cstdlib.dir/c_rbtree.c.o
[  1%] Building C object csync/src/std/CMakeFiles/cstdlib.dir/c_string.c.o
[  2%] Building C object csync/src/std/CMakeFiles/cstdlib.dir/c_time.c.o
[  2%] Linking C static library libcstdlib.a
[  2%] Built target cstdlib
Scanning dependencies of target ocsync
[  3%] Building C object csync/src/CMakeFiles/ocsync.dir/csync.c.o
[  3%] Building C object csync/src/CMakeFiles/ocsync.dir/csync_exclude.c.o
[  4%] Building C object csync/src/CMakeFiles/ocsync.dir/csync_log.c.o
[  4%] Building C object csync/src/CMakeFiles/ocsync.dir/csync_statedb.c.o
[  4%] Building C object csync/src/CMakeFiles/ocsync.dir/csync_time.c.o
[  5%] Building C object csync/src/CMakeFiles/ocsync.dir/csync_util.c.o
[  5%] Building C object csync/src/CMakeFiles/ocsync.dir/csync_misc.c.o
[  6%] Building C object csync/src/CMakeFiles/ocsync.dir/csync_update.c.o
[  6%] Building C object csync/src/CMakeFiles/ocsync.dir/csync_reconcile.c.o
[  7%] Building CXX object csync/src/CMakeFiles/ocsync.dir/csync_rename.cc.o
[  7%] Building C object csync/src/CMakeFiles/ocsync.dir/vio/csync_vio.c.o
[  8%] Building C object csync/src/CMakeFiles/ocsync.dir/vio/csync_vio_file_stat.c.o
[  8%] Building C object csync/src/CMakeFiles/ocsync.dir/vio/csync_vio_local_unix.c.o
[  9%] Linking CXX shared library libocsync.so
[  9%] Built target ocsync
Scanning dependencies of target nextcloudsync_automoc
[  9%] Automatic moc for target nextcloudsync
Generating moc source moc_abstractnetworkjob.cpp
Generating moc source moc_accessmanager.cpp
Generating moc source moc_account.cpp
Generating moc source moc_bandwidthmanager.cpp
Generating moc source moc_checksums.cpp
Generating moc source moc_clientproxy.cpp
Generating moc source moc_connectionvalidator.cpp
Generating moc source moc_cookiejar.cpp
Generating moc source moc_abstractcredentials.cpp
Generating moc source moc_dummycredentials.cpp
Generating moc source moc_httpcredentials.cpp
Generating moc source moc_discoveryphase.cpp
Generating moc source moc_excludedfiles.cpp
Generating moc source moc_logger.cpp
Generating moc source moc_networkjobs.cpp
Generating moc source moc_owncloudpropagator.cpp
Generating moc source moc_owncloudtheme.cpp
Generating moc source moc_progressdispatcher.cpp
Generating moc source moc_propagatedownload.cpp
Generating moc source moc_propagateremotedelete.cpp
Generating moc source moc_propagateremotemkdir.cpp
Generating moc source moc_propagateremotemove.cpp
Generating moc source moc_propagateupload.cpp
Generating moc source moc_propagatorjobs.cpp
Generating moc source moc_syncengine.cpp
Generating moc source moc_syncfilestatustracker.cpp
Generating moc source moc_syncjournaldb.cpp
Generating moc source moc_theme.cpp
Generating moc compilation nextcloudsync_automoc.cpp
[  9%] Built target nextcloudsync_automoc
Scanning dependencies of target nextcloudsync
[ 10%] Building CXX object src/libsync/CMakeFiles/nextcloudsync.dir/account.cpp.o
[ 10%] Building CXX object src/libsync/CMakeFiles/nextcloudsync.dir/bandwidthmanager.cpp.o
[ 11%] Building CXX object src/libsync/CMakeFiles/nextcloudsync.dir/capabilities.cpp.o
[ 11%] Building CXX object src/libsync/CMakeFiles/nextcloudsync.dir/clientproxy.cpp.o
[ 12%] Building CXX object src/libsync/CMakeFiles/nextcloudsync.dir/connectionvalidator.cpp.o
[ 12%] Building CXX object src/libsync/CMakeFiles/nextcloudsync.dir/cookiejar.cpp.o
[ 13%] Building CXX object src/libsync/CMakeFiles/nextcloudsync.dir/discoveryphase.cpp.o
[ 13%] Building CXX object src/libsync/CMakeFiles/nextcloudsync.dir/filesystem.cpp.o
[ 14%] Building CXX object src/libsync/CMakeFiles/nextcloudsync.dir/logger.cpp.o
[ 14%] Building CXX object src/libsync/CMakeFiles/nextcloudsync.dir/accessmanager.cpp.o
[ 15%] Building CXX object src/libsync/CMakeFiles/nextcloudsync.dir/configfile.cpp.o
[ 15%] Building CXX object src/libsync/CMakeFiles/nextcloudsync.dir/abstractnetworkjob.cpp.o
[ 15%] Building CXX object src/libsync/CMakeFiles/nextcloudsync.dir/networkjobs.cpp.o
[ 16%] Building CXX object src/libsync/CMakeFiles/nextcloudsync.dir/owncloudpropagator.cpp.o
[ 16%] Building CXX object src/libsync/CMakeFiles/nextcloudsync.dir/owncloudtheme.cpp.o
[ 17%] Building CXX object src/libsync/CMakeFiles/nextcloudsync.dir/progressdispatcher.cpp.o
[ 17%] Building CXX object src/libsync/CMakeFiles/nextcloudsync.dir/propagatorjobs.cpp.o
[ 18%] Building CXX object src/libsync/CMakeFiles/nextcloudsync.dir/propagatedownload.cpp.o
[ 18%] Building CXX object src/libsync/CMakeFiles/nextcloudsync.dir/propagateupload.cpp.o
[ 19%] Building CXX object src/libsync/CMakeFiles/nextcloudsync.dir/propagateremotedelete.cpp.o
[ 19%] Building CXX object src/libsync/CMakeFiles/nextcloudsync.dir/propagateremotemove.cpp.o
[ 20%] Building CXX object src/libsync/CMakeFiles/nextcloudsync.dir/propagateremotemkdir.cpp.o
[ 20%] Building CXX object src/libsync/CMakeFiles/nextcloudsync.dir/syncengine.cpp.o
[ 21%] Building CXX object src/libsync/CMakeFiles/nextcloudsync.dir/syncfilestatus.cpp.o
[ 21%] Building CXX object src/libsync/CMakeFiles/nextcloudsync.dir/syncfilestatustracker.cpp.o
[ 22%] Building CXX object src/libsync/CMakeFiles/nextcloudsync.dir/syncjournaldb.cpp.o
[ 22%] Building CXX object src/libsync/CMakeFiles/nextcloudsync.dir/syncjournalfilerecord.cpp.o
[ 23%] Building CXX object src/libsync/CMakeFiles/nextcloudsync.dir/syncresult.cpp.o
[ 23%] Building CXX object src/libsync/CMakeFiles/nextcloudsync.dir/theme.cpp.o
/tmp/client_theming/client/src/libsync/theme.cpp:33:39: fatal error: /tmp/client_theming/build-1/../nextcloudtheme/nextcloudtheme.h: No such file or directory
 #  include INCLUDE_FILE(THEME_INCLUDE)
compilation terminated.
make[2]: *** [src/libsync/CMakeFiles/nextcloudsync.dir/build.make:735: src/libsync/CMakeFiles/nextcloudsync.dir/theme.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:243: src/libsync/CMakeFiles/nextcloudsync.dir/all] Error 2
make: *** [Makefile:150: all] Error 2
```

### Client configuration
Client version: Master

Operating system: ArchLinux

OS language: en_US

I used @VGoshev's workaround to get past this error on Debian Stretch. But then I hit another roadblock during `make`:

```
/home/alex/repos/client_theming/client/src/3rdparty/certificates/p12topem.cpp: In function ‘resultP12ToPem p12ToPem(std::__cxx11::string, std::__cxx11::string)’:
/home/alex/repos/client_theming/client/src/3rdparty/certificates/p12topem.cpp:68:31: error: ‘SSLeay_add_all_algorithms’ was not declared in this scope
     SSLeay_add_all_algorithms();
                               ^
[ 29%] Building CXX object src/libsync/CMakeFiles/nextcloudsync.dir/creds/httpcredentials.cpp.o
src/libsync/CMakeFiles/nextcloudsync.dir/build.make:950: recipe for target 'src/libsync/CMakeFiles/nextcloudsync.dir/__/3rdparty/certificates/p12topem.cpp.o' failed
make[2]: *** [src/libsync/CMakeFiles/nextcloudsync.dir/__/3rdparty/certificates/p12topem.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
```

--- a/README.md
+++ b/README.md
@@ -17,7 +17,7 @@ Run:
 ```bash
 mkdir build-linux
 cd build-linux
-cmake -D OEM_THEME_DIR=$(realpath `pwd`/../nextcloudtheme)  ../client
+cmake -D OEM_THEME_DIR=$(realpath ../nextcloudtheme)  ../client
 make
 make install
 ```

--- a/README.md
+++ b/README.md
@@ -17,7 +17,7 @@ Run:
 ```bash
 mkdir build-linux
 cd build-linux
-cmake -D OEM_THEME_DIR=`pwd`/../nextcloudtheme ../client
+cmake -D OEM_THEME_DIR=$(realpath `pwd`/../nextcloudtheme)  ../client
 make
 make install
 ```

